### DISCUSSION TITLE
ImageMagick install on old MacOS Server for PHP?

### DISCUSSION THREAD

**Discussion by tcmeyers** [ImageMagick/ImageMagick#5722] [General]
I have a web server running on macOS Sierra 10.12.6 with the old Apple "Server" app 5.3.1. I have purposely not moved on because later Server dropped features I need (the Web Server part).

My pages are heavily PHP based, again, old, 5.6.30.

With larger photos being submitted to me, GD doesn't seem to handle many, and I saw a suggestion that ImageMagick would do better. But I know little about installing things properly.

Can anyone help me out for this specific situation? I'd appreciate it a lot.

Troy

**Comment by fmw42** [comment]
When you say PHP are you talking about PHP exec() or PHP Imagick? PHP Imagick requires Imagemagick and has its own command structure that makes calls to Imagemagick. They are not the same. PHP exec() just uses Imagemagick command line commands as if typed in a terminal.

I know little about installing PHP Imagick.  You can install Imagemagick from the relevant binary on Mac OSX, if you can find a compatible version for Sierra.  Imagemagick library of installs does not keep old versions.  But you can find binary installs from Homebrew and also from MacPorts.  Alternately, you can install all your needed delegates from MacPorts and install Imagemagick from source. Source files are archived if you need an older version. If you desire the latter, then I have some instructions that I used for the Mac to do that. But you would be better to find a binary from either Homebrew or MacPorts for installation.  See https://imagemagick.org/script/download.php#macosx

**Reply by tcmeyers** [reply]
Thanks, Fred!

I guess I will just go for broke and try it.

I clearly don't know what I am talking about. I want to be able to call ImageMagick functions from within PHP on PHP structure, with PHP being executed by the web server in order to generate response pages for web-based queries. Usually from web remote web browsers.

Specifically I'm hoping to replace the PHP functions:
imagecreatetruecolor to create an instance of an image object
imagecopyresized to copy a portion and resize an image pulled out of a database or server file
imagecopyresampled to do the same as imagecopyresized but with interpolation
imagedestroy to free up memory used

I need it to work on very large JPGs to produce thumbnails and scaled-down but largish images on the fly. PHP's GD library seems to fail at a certain image size, wasn't a problem until recently, with larger and larger photos.

So, I'm not really sure what I want, but I think it's the PHP Imagick class, as described here:

https://www.php.net/manual/en/book.imagick.php

Thanks again.

**Reply by fmw42** [reply]
Unfortunately, I know little about installing PHP Imagick. But first you have to install Imagemagick. Or perhaps the Imagick install does that for you.  

But see https://ma.ttias.be/install-phps-imagick-extension-on-mac-with-brew/

**Comment by namrehcram** [comment]
Hi Troy, 
I'm in the exact same situation as you are. 
Were you able to solve the issue?
